src/components/screens/SendScreen/SendSearchContacts.tsx

```diff
@@ -60,39 +76,76 @@ const SendSearchContacts: React.FC<SendSearchContactsProps> = ({
     resetSendRecipient();
   }, [loadRecentAddresses, resetSendRecipient]);
 
+  // Clear QR data when component unmounts
+  useEffect(() => () => clearQRData(), [clearQRData]);
+
   /**
    * Handles search input changes and updates suggestions
    *
    * @param {string} text - The search text entered by user
    */
-  const handleSearch = (text: string) => {
-    setAddress(text);
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    searchAddress(text);
-  };
+  const handleSearch = useCallback(
+    (text: string) => {
+      setAddress(text);
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      searchAddress(text);
+    },
+    [searchAddress],
+  );
 
   /**
    * Handles when a contact or address is selected
    *
    * @param {string} contactAddress - The selected contact address
    */
-  const handleContactPress = (contactAddress: string) => {
-    if (recentAddresses.some((c) => c.address === contactAddress)) {
-      analytics.track(AnalyticsEvent.SEND_PAYMENT_RECENT_ADDRESS);
-    }
-    // Save to both stores for different purposes
-    // Send store is for contact management
-    setDestinationAddress(contactAddress);
-    // Transaction settings store is for the transaction flow
-    saveRecipientAddress(contactAddress);
+  const handleContactPress = useCallback(
+    (contactAddress: string) => {
+      if (recentAddresses.some((c) => c.address === contactAddress)) {
+        analytics.track(AnalyticsEvent.SEND_PAYMENT_RECENT_ADDRESS);
+      }
+      // Save to both stores for different purposes
+      // Send store is for contact management
+      setDestinationAddress(contactAddress);
+      // Transaction settings store is for the transaction flow
+      saveRecipientAddress(contactAddress);
 
-    navigation.goBack();
-  };
+      navigation.goBack();
+    },
+    [recentAddresses, setDestinationAddress, saveRecipientAddress, navigation],
+  );
 
   const handlePasteFromClipboard = () => {
     getClipboardText().then(handleSearch);
   };
 
+  const handleOpenQRScanner = () => {
+    // Navigate to the root navigator's QR scanner screen
+    navigation.navigate(ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN, {
+      source: QRCodeSource.ADDRESS_INPUT,
+    });
+  };
+
+  // Handle scanned QR data when available
+  useEffect(() => {
+    if (
+      scannedData &&
+      storedSource === QRCodeSource.ADDRESS_INPUT &&
+      !isConsumed
+    ) {
+      // Use the scanned data as the search input
+      handleSearch(scannedData);
+      // Mark the data as consumed
+      consumeQRData();
+    }
+  }, [scannedData, storedSource, isConsumed, consumeQRData, handleSearch]);
```

Comment:
do we still need to handle this here in case we navigate users straight to the Amount screen as mentioned [here](https://github.com/stellar/freighter-mobile/pull/321#issuecomment-3246659361)?